snapcraft/internal/repo/snaps.py

```diff
@@ -116,7 +132,13 @@ def get_current_channel(self):
 
     def is_classic(self):
         store_channels = self._get_store_channels()
-        return store_channels[self.channel]['confinement'] == 'classic'
+        try:
+            return store_channels[self.channel]['confinement'] == 'classic'
+        except KeyError:
+            logger.debug('Current store channels are {!r} and the store'
```

Comment:
Does this ever happen when it isn't a legit error? Might be helpful to actually print this like an error (but formatted a little better), which would be nice for determining the channels that _are_ available in case they mistyped one or the snap isn't released into that channel.

Reply:
we don't know why this happens, we see it in the armhf tests, this PR just started as printing this log then I discovered something was wrong with the logic that verifies we were doing the right thing (as was mentioned in part of the docstring).